README.md

```diff
@@ -1,277 +1,899 @@
+    - `pairBeacon` == pair beacon asset name for this swap.
+    - `offerId` == asset_id of the offer asset.
+    - `offerName` == asset_name of the offer asset.
+    - `offerBeacon` == offer beacon asset name for this swap's offer asset.
+    - `askId` == asset_id of the ask asset.
+    - `askName` == asset_name of the ask asset.
+    - `swapPrice` denominator > 0
+    - `swapPrice` > 0
+5) The offer asset and ask assets must be different assets.
+6) The address' staking credential must signal approval.
+7) Any unused beacons must be burned.
+
+This redeemer can be used with either beacon redeemer since both allow burning. If beacons only need
+to be burned, it is cheaper to use the `BurnBeacons` redeemer. However, if even a single beacon must
+be minted, the `CreateSwap` redeemer must be used. This behavior enables the swap owner to change
+what trading pair a swap is for in a single transaction. This is instead of having to close the
+original swap in one transaction only to re-open it in another transaction.
+
+Requirement 6 guarantees that only the owner can close/update a swap and claim the assets.
+
+##### Executing a Swap
+
+Any Cardano user can execute an available swap using the `Swap` Redeemer as long as the swap
+conditions are met.
+
+At a high level, the `Swap` redeemer uses the swap input's output reference and datum to find the
+corresponding swap output. The checks are essentially:
+1) Does this output have the trading pair beacon from the input?
+2) If "Yes" to (1), is this output locked at the address where the input comes from?
+3) If "Yes" to (2), does this output have the proper datum for the corresponding output?
+4) If "Yes", this is the corresponding output.
+
+It then compares the value of that output to the input to determine the swap's asset flux. Since the
+validator first checks for the trading pair beacon, each execution is dedicated to a specific
+trading pair. Any other outputs are ignored in this specific execution. This logic works because a
+script is executed once for every UTxO spent from the address. If input 1 is for beacon XYZ and
+input 2 is for beacon ABC, the first execution can be dedicated to beacon XYZ and the second
+execution can be dedicated to ABC. The net transaction will only succeed if all executions succeed.
+This behavior allows cheaply composing swaps of different trading pairs that are located at the same
+address. In other words, the design is taking advantage of the redundant executions.
+
+At a low-level, for a swap execution to be successfull, all of the following must be true:
+
+1. The input must contain the beacon for that trading pair - the required beacon is gotten from the
+   datum for that UTxO. If the beacon is present, that means the datum can be trusted.
+2. There must be an output to this address with the proper value and inline `SwapDatum`:
+    - Must contain exactly the same value as the input except for the ask asset, offer asset, and
+    ADA.
```

Comment:
yes, exchange rates or as a similar exchange symbol like currency quotes and values.  Generally it is capitalised because it's symbolic.  Here's our operative explanation on the Developer Portal (which unfortunately doesn't include Ada for the crypto itself: e.g. "Lebanon parliament voted to make Ada legal tender"): https://developers.cardano.org/docs/portal-style-guide/#editorial-style-guide